### PR TITLE
[13824] add check_xml_static_discovery(std::string& xml)

### PR DESCRIPTION
**Pull request by inesluno**
Add a member function `ReturnCode_t DomainParticipantFactory::check_xml_static_discovery(std::string& xml_file)` that can be used to check Static Discovery XML by calling `XMLEndpointParser::loadXMLFile()`

Signed-off-by: Ines Luño <inesluno@eprosima.com>

### CODE REVIEW

File: src/cpp/fastdds/domain/DomainParticipantFactory.cpp [hunk 1]
```diff
@@ -386,6 +387,18 @@ ReturnCode_t DomainParticipantFactory::load_XML_profiles_string(
     return ReturnCode_t::RETCODE_OK;
 }
 
+ReturnCode_t DomainParticipantFactory::check_xml_static_discovery(
+    std::string& xml_file)
+{
+    XMLEndpointParser parser;
+    if (XMLP_ret::XML_ERROR == parser.loadXMLFile(xml_file))
```
Comment (EduPonz) [inline src/cpp/fastdds/domain/DomainParticipantFactory.cpp:394]: Right now, the parser only returns either error or ok, but that may change in the future. In that case, I'd only return ok whenever the parser does so.
```suggestion
    if (XMLP_ret::XML_OK != parser.loadXMLFile(xml_file))
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
1. Please address warnings
2. Uncrustify failed
3. I'd separate the Doxygen related changes into a separate PR that we can merge for v2.5.1
4. I'm missing tests for the new API

**Review by EduPonz — CHANGES_REQUESTED** [review]
Please, document this new API in [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/static.html#static-discovery-settings)

File: src/cpp/fastdds/domain/DomainParticipantFactory.cpp (hunk 1 above)
Comment (inesluno) [reply src/cpp/fastdds/domain/DomainParticipantFactory.cpp:394]: ok, committed

**Comment by EduPonz** [comment]
Closed in favor of #2817

**Pull request closed without merging**